components/sup/src/package/mod.rs

```diff
@@ -233,21 +234,16 @@ impl Package {
     /// helpers above.
     pub fn config_files(&self) -> Result<Vec<String>> {
         let mut files: Vec<String> = Vec::new();
-        match std::fs::read_dir(self.path().join("config")) {
-            Ok(config_dir) => {
-                for config in config_dir {
-                    let config = try!(config);
-                    match config.path().file_name() {
-                        Some(filename) => {
-                            files.push(filename.to_string_lossy().into_owned().to_string());
-                        }
-                        None => unreachable!(),
-                    }
+        if std::fs::metadata(self.path().join("config")).unwrap().is_dir() {
+            for entry in WalkDir::new(self.path().join("config")) {
+                let entry = entry.unwrap();
+                if entry.path().is_file() {
+                    let rel_path = entry.path().strip_prefix(&self.path().join("config").to_str().unwrap()).unwrap().to_string_lossy().into_owned().to_string();
```

Comment:
we try to handle as many `Result` values with pattern matching. In general, I'm suspicious whenever I see an unhandled `unwrap()` in the wild.

I took a shot at refactoring this method a bit:

```
 pub fn config_files(&self) -> Result<Vec<String>> {
        let mut files: Vec<String> = Vec::new();
        let cfg_dir = self.path().join("config");

        match std::fs::metadata(&cfg_dir).map(|d| d.is_dir()) {
            Err(_) | Ok(false) => {
                // error reading config dir or dir doesn't exist
                return Ok(files)
            }
            Ok(true) => ()
        };

        for entry in WalkDir::new(&cfg_dir) {
            if let Err(e) = entry {
                debug!("Can't process config file {}", e);
                continue;
            }
            let entry = entry.unwrap();
            if entry.path().is_file() {
                if let Ok(rel_path) = entry.path().strip_prefix(&cfg_dir) {
                    let f = rel_path.to_string_lossy().into_owned();
                    println!("Detected config file = {}", &f);
                    files.push(f);
                }
            }
        }
        Ok(files)
    }
```
In the case of `entry` above, to reduce the amount of blocks + indentation, I check to see if `entry` is an `Err` and move on, or unwrap if it's not. This is circumventing the compilers exhaustive pattern matching, but I've used this technique carefully in other spots to reduce too many nested `{ }`'s.